src/Stratis.Bitcoin.Features.PoA/PoAConsensusRuleEngine.cs

```diff
@@ -18,6 +18,8 @@ public class PoAConsensusRuleEngine : PowConsensusRuleEngine
     {
         public ISlotsManager SlotsManager { get; private set; }
 
+        public IFederationHistory FederationHistory { get; private set; }
```

Comment:
@quantumagi I want us to move away from injecting components into the rule engine like this. As the rules are now DI-ed anyway, can you please move this to the rule's constructor and inject it there?

Reply:
This is passed via the constructor. Can you clarify?

Reply:
What I mean is, inject this directly into the rule's constructor, not the rule engine.